master/master.go

```diff
@@ -126,3 +144,37 @@ func instanceIds(instances []*ec2.Instance) []*string {
 	}
 	return instanceIds
 }
+
+func (m *Master) getTaskToSchedule() int {
+	for idx, t := range m.newTasks {
+		if !m.jobs[t.JobID].reachedMaxTasks() {
+			return idx
+		}
+	}
+	return -1
+}
+
+func (m *Master) slotsAvailable() bool {
+	for _, i := range m.instances {
+		if len(i.tasks) < i.maxTasks {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *Master) scheduleTask(tidx int) {
+	minResources := math.MaxInt64
+	var slaveIP string
+	for k, v := range m.instances {
+		if len(v.tasks) < minResources {
```

Comment:
The `break` is not possible, because we are looking for the slave that is currently the less occupied. 

@kc1212 You are right, MMR +50. I am so noob.